tests/SpecTests/ResultExpectation.php

```diff
@@ -274,6 +287,10 @@ public function assert(TestCase $test, $actual)
                 $test->assertSameDocuments($expected, $actual);
                 break;
 
+            case self::ASSERT_DOCUMENTS_MATCH:
+                $test->assertDocumentsMatch($expected, $actual);
```

Comment:
I initially thought this might be redundant in light of `ASSERT_MATCHES_DOCUMENT`, but I remembered that they're actually different. `MongoDB\Tests\SpecTests\FunctionalTestCase::assertDocumentsMatch()` uses DocumentsMatchConstraint, configured to ignore extra keys in both root and embedded documents. `ASSERT_MATCHES_DOCUMENT` does a quick type check on both arguments (expected and actual) and then delegates to `MongoDB\Tests\TestCase::assertMatchesDocument()`, which re-orders fields and manually unsets extra keys in the actual value before converting both documents to JSON and doing a string comparison.

We already have [PHPLIB-436](https://jira.mongodb.org/browse/PHPLIB-436) in our backlog to refactor `assertMatchesDocument` to use DocumentsMatchConstraint so I don't think we need a new comment in this code to reference that; however, I'll add a note in the JIRA issue to consolidate `ASSERT_MATCHES_DOCUMENT` and `ASSERT_DOCUMENTS_MATCH` down the line.